### ISSUE TITLE
Better support for a simple spinner

### ISSUE THREAD

**Issue by jennybc** [r-lib/progress#81]
I'm using C++ progress from this package in readxl now.

In my first pass, I used separate progress bars for each of several steps in processing a worksheet. In some steps, I don't actually know how much work lies ahead of me. In others, I sort of do but it's annoying to compute and still approximate. In any case, I just want a plain old spinner to comfort people while they wait.

In my second pass, I will use one progress bar for all the step and now I truly can't compute how much work there is to do.

I created my own `Spinner` class for this (over in readxl) but it would be nice if this package offered more for this use case.

**Comment by gaborcsardi** [comment]
Yes, there is a spinner in the R API, the C++ API lacks several features, unfortunately.

**Comment by gaborcsardi** [comment]
I believe this is possible with cli progress bars already, so I am going to close this issue.
